core/src/avm2/globals/flash/media/sound.rs

```diff
@@ -279,7 +283,96 @@ pub fn load_compressed_data_from_byte_array<'gc>(
     let handle = activation.context.audio.register_mp3(bytes).map_err(|e| {
         Error::RustError(format!("Failed to register sound from bytearray: {e:?}").into())
     })?;
-
+    let progress_evt = activation
```

Comment:
Is this event really dispatched for `loadCompressedDataFromByteArray`? If so, can you check that the `complete` event is also dispatched? Also, might it be worth it extracting the ID3 logic into a function in `SoundObject`?

Reply:
I've checked that, only progress and id3 event are called, complete event isn't called.

Reply:
Okay, I've added a read_and_call_id3_event function and used it instead.